### PR TITLE
Publicize AssetCheckSummaryRecord + instance method

### PR DESCRIPTION
**Pull request by dpeng817**
Publicize AssetCheckSummaryRecord and kin. This API is quite useful for creating sensors that might make use of an asset check's result. See: https://github.com/dagster-io/dagster/discussions/22176

### CODE REVIEW

**Comment by dpeng817** [comment]
* **#22678** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22678?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by sryza — COMMENTED** [review]
We should make this experimental for now.

@prha - do you have any reservations about making `AssetCheckSummaryRecord` public?

**Comment by prha** [comment]
>  do you have any reservations about making AssetCheckSummaryRecord public?

No, other than the fact that `SummaryRecord` suffix doesn't mean anything to anyone right now.  We will probably want/need to rename `AssetRecord` => `AssetSummaryRecord` or something like that.

File: python_modules/dagster/dagster/_core/storage/event_log/base.py [hunk 1]
```diff
@@ -135,6 +135,12 @@ class AssetRecord(NamedTuple):
 
 
 class AssetCheckSummaryRecord(NamedTuple):
+    """A representation of the current state of an asset check. Contains the most recent execution record,
+    and the most recent run id that the check was executed in.
+
+    Users should not invoke this class directly.
```
[the remark this answers is not in the document]
Comment (dpeng817) [reply python_modules/dagster/dagster/_core/storage/event_log/base.py:141]: heads up that we use `invoke` terminology for other classes here; see `PlannedMaterializationInfo` for ex.

[the remark this answers is not in the document]
Comment (dpeng817) [reply python_modules/dagster/dagster/_core/storage/event_log/base.py:141]: that said, I made that change for this class.

**Comment by dpeng817** [comment]
@prha agreed that would be a good followup.

**Comment by sryza** [comment]
@schrockn - I recall you had some thoughts about the `SummaryRecord` naming here? I think this is in line with the direction you were advocating for, but was relatively farther away from this discussion, so wanted to loop you in.

File: python_modules/dagster/dagster/_core/storage/asset_check_summary_record.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from typing import NamedTuple, Optional
+
+from dagster._annotations import experimental
+from dagster._core.definitions.asset_check_spec import AssetCheckKey
+from dagster._core.storage.asset_check_execution_record import AssetCheckExecutionRecord
+
+
+@experimental
+class AssetCheckSummaryRecord(NamedTuple):
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/storage/asset_check_summary_record.py:9]: go for `@record`?

**Review by schrockn — APPROVED** [review]
👍🏻 gtg for the name from my end

**Review by sryza — APPROVED** [review]

**Comment by dpeng817** [comment]
Closing this until such a time when I get around to it again

**Pull request closed without merging**